### PR TITLE
Peru 1.1.4 (new formula)

### PR DESCRIPTION
**Pull request by colindean**
Fixes buildinspace/peru#130

- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Peru is a dependency retriever that can include other VCS repos and archives within a project's build dependencies in a VCS and build system-agnostic way.

### CODE REVIEW

**Comment by colindean** [comment]
I used https://github.com/Homebrew/homebrew-core/blob/ad166521867512219d4043ed97dec25ed66834bb/Formula/rtv.rb for inspiration.

**Comment by DomT4** [comment]
> I used https://github.com/Homebrew/homebrew-core/blob/ad166521867512219d4043ed97dec25ed66834bb/Formula/rtv.rb for inspiration.

I'm not keen to roll this out as the universal standard for handling Python resources at this point in time. Please instead use manual `resource do` blocks, which you can obtain easily using `homebrew-pypi-poet`, available via `pip`, and then you should just be able to make `def install` as simple as `virtualenv_install_with_resources`.

**Comment by colindean** [comment]
Dropping some things in here for me to come back to this this week:

https://github.com/tdsmith/homebrew-pypi-poet
https://pypi.org/project/homebrew-pypi-poet/

**Comment by colindean** [comment]
@DomT4 Are these good examples of what the result should look like?

* [tox](https://github.com/Homebrew/homebrew-core/blob/81e634d3d54bd00ef9fdb9e4a6e21bb0a9f557e9/Formula/tox.rb)
* [dcos-cli](https://github.com/Homebrew/homebrew-core/blob/5381f99342031e48ce33583bb766f0e2db8b7be3/Formula/dcos-cli.rb)

**Comment by DomT4** [comment]
Yup! `tox` there is pretty much the standard we _try_ to hit where possible.

**Comment by colindean** [comment]
Not that I'm asking for more stuff to do, but what importance would you place on conforming python packages to that standard? As in, what is to be gained by moving `rtv` to that structure?

**Comment by DomT4** [comment]
It's a difficult thing to roll out one standard globally, sadly.

Some things have so many Python-based dependencies, which we stage as resources, that it makes updating a formula beyond painful when any significant period of time between releases has elapsed, and in those cases it can be tempting to let `pip` do the heavy lifting for us.

It's a constant balance between trying to retain control over the build process whilst also not making formulae so contributor-hostile that it always falls on maintainers to handle the updates. Most things have few enough resources to allow us to retain control over the entire build process, thankfully, and that's our preference.

In time it's likely/possible we'll harmonise on one style more universally, but for now it's basically trying to hit that balance.

**Comment by colindean** [comment]
I haven't forgotten about this, going to try to get to it tonight or this weekend.

**Comment by colindean** [comment]
@DomT4, I've switched this formula to poet output!

File: Formula/peru.rb
Comment (commitay) [inline Formula/peru.rb]: This line is can be removed, `virtualenv_install_with_resources` does this automatically.

Comment (colindean) [reply Formula/peru.rb]: 👍 I guess poet should be updated, then? https://github.com/tdsmith/homebrew-pypi-poet/blob/fdafc615bcd28f29bcbe90789f07cc26f97c3bbc/poet/templates.py#L32

Comment (commitay) [reply Formula/peru.rb]: https://github.com/tdsmith/homebrew-pypi-poet/pull/61, https://github.com/tdsmith/homebrew-pypi-poet/issues/63

File: Formula/peru.rb
Comment (commitay) [inline Formula/peru.rb]:
```ruby
    (testpath/"peru.yaml").write <<~EOS
      imports:
        peru: peru
      git module peru:
        url: https://github.com/buildinspace/peru.git
    EOS
```

File: Formula/peru.rb
Comment (commitay) [inline Formula/peru.rb]: Can we assert some output from this?

Comment (colindean) [reply Formula/peru.rb]: Yes. I've asserted the existence of both the `.peru` directory and the `peru` directory that should contain the git clone that peru… clones.

File: Formula/peru.rb
Comment (colindean) [reply Formula/peru.rb]: 👍

**Review by DomT4 — APPROVED** [review]

**Comment by commitay** [comment]
@DomT4 Is this okay to 🚢 now? It's still labeled <kbd>do not merge</kbd>.

**Review by DomT4 — APPROVED** [review]

**Comment by DomT4** [comment]
@commitay Apologies! Label lifted. 🚢 when desired.

**Comment by commitay** [comment]
@DomT4 No problem, thanks for the quick response!

@colindean Thanks for the 🆕 formula!

**Comment by colindean** [comment]
🎉

**Pull request closed without merging**
